Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -255,6 +255,10 @@ void LocalFrameViewLayoutContext::performLayout(bool canDeferUpdateLayerPosition
             showRenderTree(renderView());
 #endif
     }
+
+    if (document())
+        Style::AnchorPositionEvaluator::updateAnchorPositioningStatesAfterInterleavedLayout(*document());
```

Comment:
This is needed because `Scope::invalidateForAnchorDependencies` makes a reverse map from anchor -> anchored elements. This depends on `AnchorPositionedState::anchorElements` being populated, and this is populated by `Style::AnchorPositionEvaluator::updateAnchorPositioningStatesAfterInterleavedLayout`. So this method has to be called at least once before `Scope::invalidateForAnchorDependencies` runs, otherwise no anchored elements will be invalidated.

@anttijk I'm not sure if this is the correct solution to the problem, or it's just a hack.

Reply:
At least it does not have correct name now because it is called after every layout, not just interleaved layout.

I think there needs to be a clear plan on how this actually should work.